test/bugs/issue_543.test.ts

```diff
@@ -19,21 +19,21 @@ describe("Issue 543", () => {
 
         @injectable()
         class Child2 {
-            public circ: Circular;
+            public circ: unknown;
```

Comment:
do we need unknown here ?

Reply:
I think yes. If you try to run the tests targeting ES6 and you do not change typings, you get the folloing error:

```
ReferenceError: Cannot access 'Circular' before initialization
      at Context.<anonymous> (test/bugs/issue_543.test.ts:24:46)
      at processImmediate (internal/timers.js:456:21)
```

I think we can change the type because at the end it's sugar syntax. What we really want to check is we are able to detect a circular dependency but maybe I'm missing something

Reply:
I was working on this yesterday and rewrote the tests I will try to add just that part